AssetGenerator/Source/AssetGenerator/Private/Toolkit/AssetTypeGenerator/MaterialGenerator.cpp

```diff
@@ -670,20 +682,55 @@ void DisconnectIfExpressionMissing(FExpressionInput& Input, UMaterial* Material)
 	}
 }
 
+static bool PreferEmissiveColor(EMaterialDomain Domain) {
+	return Domain == EMaterialDomain::MD_PostProcess || Domain == EMaterialDomain::MD_UI;
+}
+
 void UMaterialGenerator::TryConnectBasicMaterialPins(UMaterial* Material) {
 	FExpressionInput& BaseColorInput = Material->GetEditorOnlyData()->BaseColor;
 	FExpressionInput& NormalInput = Material->GetEditorOnlyData()->Normal;
+	FExpressionInput& EmissiveColorInput = Material->GetEditorOnlyData()->EmissiveColor;
 
 	DisconnectIfExpressionMissing(BaseColorInput, Material);
 	DisconnectIfExpressionMissing(NormalInput, Material);
+	DisconnectIfExpressionMissing(EmissiveColorInput, Material);
+
+	if (Material->MaterialDomain == EMaterialDomain::MD_PostProcess) {
+		const FVector2D NodePos = FVector2D(Material->EditorX - 384, Material->EditorY);
+		// This terribleness is necessary because UMaterialExpressionSceneTexture
+		// is not exported at all, so there is no way to use the type directly...
+		UPackage* EngineScriptPackage = UEngine::StaticClass()->GetOuterUPackage();
+		TSubclassOf<UMaterialExpression> SceneTextureExpressionClass = FindObjectChecked<UClass>(EngineScriptPackage, TEXT("MaterialExpressionSceneTexture"));
+		UMaterialExpression* SceneTextureExpression = SpawnMaterialExpression<UMaterialExpression>(Material, NodePos, SceneTextureExpressionClass);
+		if (FByteProperty* Prop = CastField<FByteProperty>(SceneTextureExpressionClass->FindPropertyByName(TEXT("SceneTextureId")))) {
+			if (uint8* ValuePtr = Prop->ContainerPtrToValuePtr<uint8>(SceneTextureExpression)) {
+				*ValuePtr = PPI_PostProcessInput0;
+			} else {
+				fgcheckf(0, TEXT("bad ValuePtr"));
+			}
+		} else {
+			fgcheckf(0, TEXT("bad Prop"));
+		}
```

Comment:
The terribleness can be replaced by a milder terribleness

```suggestion
		UPackage* EngineScriptPackage = FindPackage(NULL, UMaterialExpressionSceneTexture::StaticPackage());
		TSubclassOf<UMaterialExpression> SceneTextureExpressionClass = FindObjectChecked<UClass>(EngineScriptPackage, TEXT("MaterialExpressionSceneTexture"));
		UMaterialExpressionSceneTexture* SceneTextureExpression = static_cast<UMaterialExpressionSceneTexture*>(SpawnMaterialExpression<UMaterialExpression>(Material, NodePos, SceneTextureExpressionClass));
		SceneTextureExpression->SceneTextureId = PPI_PostProcessInput0;
```